php7.2-stretch/Dockerfile

```diff
@@ -0,0 +1,174 @@
+FROM php:7.2-fpm-stretch AS build
+
+ENV PATH=/root/.local/bin:$PATH \
+    PROCFILE_PATH=/app/Procfile \
+    LC_ALL=C.UTF-8 \
+    NVM_DIR=/opt/nvm \
+    NVM_VERSION=0.34.0
+
+USER root
+
+RUN echo "force-unsafe-io" > /etc/dpkg/dpkg.cfg.d/02apt-speedup
+RUN echo "Acquire::http {No-Cache=True;};" > /etc/apt/apt.conf.d/no-cache
+
+RUN DEBIAN_FRONTEND=noninteractive dpkg-reconfigure -f noninteractive tzdata
+RUN apt-get update && apt-get upgrade -y
+
+RUN DEBIAN_FRONTEND=noninteractive apt-get install -yqq \
+      apt-transport-https \
+      apt-utils \
+      build-essential \
+      curl \
+      dumb-init \
+      gnupg2 \
+      libc-client-dev \
+      nginx \
+      openssh-client \
+      unzip \
+      zlib1g-dev \
+      --no-install-recommends \
+  && rm -rf /var/lib/apt/lists/*
+
+ENV buildDependencies=" \
+    libbz2-dev \
+    libsasl2-dev \
+    "
+
+ENV runtimeDependencies=" \
+    imagemagick \
+    libfreetype6-dev \
+    libicu-dev \
+    libjpeg62-turbo-dev \
+    libjpeg-dev \
+    libkrb5-dev \
+    libldap2-dev \
+    libmagickwand-dev \
+    libmemcached-dev \
+    libmemcachedutil2 \
+    libpng-dev \
+    libpq-dev \
+    libssl-dev \
+    libwebp-dev \
+    libxml2-dev \
+    libxpm-dev \
+    libzip-dev \
+    "
+
+ENV extensions=" \
+    bcmath \
+    bz2 \
+    calendar \
+    exif \
+    iconv \
+    intl \
+    mbstring \
+    opcache \
+    pcntl \
+    pdo_pgsql \
+    pgsql \
+    soap \
+    xml \
+    xmlrpc \
+    zip \
+    "
+
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get -yqq install $buildDependencies --no-install-recommends \
+  && DEBIAN_FRONTEND=noninteractive apt-get -yqq install $runtimeDependencies --no-install-recommends \
+  && rm -rf /var/lib/apt/lists/* \
+  && docker-php-ext-install -j$(nproc) $extensions \
+  && docker-php-ext-configure gd --with-gd --with-webp-dir --with-jpeg-dir \
+    --with-png-dir --with-zlib-dir --with-xpm-dir --with-freetype-dir \
+    --enable-gd-native-ttf \
+  && docker-php-ext-install -j$(nproc) gd \
+  && docker-php-ext-configure ldap --with-libdir=lib/x86_64-linux-gnu/ \
+  && docker-php-ext-install -j$(nproc) ldap \
+  && docker-php-ext-configure imap --with-kerberos --with-imap-ssl \
+  && docker-php-ext-install -j$(nproc) imap
+
+RUN docker-php-source extract \
+  && curl -L -s https://github.com/php-memcached-dev/php-memcached/archive/v3.1.3.zip -o /tmp/php-memcached-3.1.3.zip \
+  && unzip /tmp/php-memcached-3.1.3.zip -d /tmp \
+  && mv /tmp/php-memcached-3.1.3 /usr/src/php/ext/memcached \
+  && docker-php-ext-install memcached \
+  && docker-php-ext-enable memcached \
+  && docker-php-source delete
+
+RUN pecl channel-update pecl.php.net \
+  && pecl install imagick redis xdebug \
+  && docker-php-ext-enable imagick redis xdebug
+
+RUN echo "memory_limit=1G" > /usr/local/etc/php/conf.d/zz-conf.ini
```

Comment:
Why 1G? What does it depend on?
Maybe we need to make this configurable.

Reply:
This is a default value big enough not to crash on installing. However, the value can be (and is) set in the configuration inside the boilerplates for each project. 
The default is 128M which is sometimes too low and makes composer installations crash